client/src/main/java/org/eclipse/hono/client/impl/ProtocolAdapterCommandConsumerFactoryImpl.java

```diff
@@ -195,11 +202,19 @@ protected void onDisconnect() {
                 });
     }
 
-    private Future<Void> removeCommandConsumer(final String tenantId, final String deviceId,
-            final SpanContext onCloseSpanContext) {
-        log.trace("remove command consumer [tenant-id: {}, device-id: {}]", tenantId, deviceId);
-        adapterInstanceCommandHandler.removeDeviceSpecificCommandHandler(tenantId, deviceId);
+    private Future<Void> removeCommandConsumer(final CommandHandlerWrapper commandHandlerWrapper, final Duration lifespan,
+            final Instant lifespanStart, final SpanContext onCloseSpanContext) {
+
+        final String tenantId = commandHandlerWrapper.getTenantId();
+        final String deviceId = commandHandlerWrapper.getDeviceId();
 
+        log.trace("remove command consumer [tenant-id: {}, device-id: {}]", tenantId, deviceId);
+        if (!adapterInstanceCommandHandler.removeDeviceSpecificCommandHandler(commandHandlerWrapper)) {
```

Comment:
this is the check for scenario B, right (device has re-connected to same verticle instance)? IMHO it would be very helpful to describe this scenario in a comment here